### PR TITLE
(MODULES-4818) Fix iis_feature spec

### PR DESCRIPTION
**Pull request by jpogran**
This renames the iis_feature spec file so that it's actually picked up by the test runs. It also fixes the iis_feature detection so that it correctly returns true of false if the feature is an IIS feature or not

### CODE REVIEW

**Comment by glennsarti** [comment]
Running through adhoc in case the tests are broken.

**Comment by glennsarti** [comment]
The acceptance tests are failing in smoke
```
1:23:21 Finished in 17 minutes 9 seconds (files took 1 minute 24.13 seconds to load)
11:23:21 127 examples, 3 failures
11:23:21 
11:23:21 Failed examples:
11:23:21 
11:23:21 rspec ./spec/acceptance/iis_feature_spec.rb[1:1:1:1:1] # iis_feature when managing features with default parameters behaves like an idempotent resource should run without errors
11:23:21 rspec ./spec/acceptance/iis_feature_spec.rb[1:1:1:1:2] # iis_feature when managing features with default parameters behaves like an idempotent resource should run a second time without changes
11:23:21 rspec ./spec/acceptance/iis_feature_spec.rb:17 # iis_feature when managing features with default parameters when puppet resource is run should report the correct ensure value
```
Errors are:
```
 undefined method `includes?' for #<Array:0x000000048d8f70>
```
so it looks like you'll need cherry pick the fixes from https://github.com/puppetlabs/puppetlabs-iis/pull/62

**Review by glennsarti — DISMISSED** [review]
Needs fixes from https://github.com/puppetlabs/puppetlabs-iis/pull/62

**Comment by jpogran** [comment]
Merged in #62 and repushed

**Comment by glennsarti** [comment]
The iis_feature tests reference `iis_feature { 'Web-Asp-Net45':` which does not exist as a valid feature on 2008R2 or apparently Windows Server 2012R2

Could a type issue

Errors with
```
12:44:43   Info: Loading facts
12:44:52   Notice: Compiled catalog for qwdlk25w9kijgdf.delivery.puppetlabs.net in environment production in 0.20 seconds
12:44:52   Info: Applying configuration version '1492803891'
12:44:52   Error: iis_feature can only be used to install IIS features. 'Web-Asp-Net45' is not an IIS feature
12:44:57   Error: /Stage[main]/Main/Iis_feature[Web-Asp-Net45]/ensure: change from absent to present failed: iis_feature can only be used to install IIS features. 'Web-Asp-Net45' is not an IIS feature
12:44:57   Notice: Applied catalog in 5.41 seconds
12:44:57 
12:44:58 qwdlk25w9kijgdf.delivery.puppetlabs.net (windows2012r2-64-1) executed in 14.70 seconds
12:44:58 Exited: 4
```

**Comment by glennsarti** [comment]
Perhaps use `Web-Asp-Net` to be safe across 2008 to 2016.  Not sure why it's throwing that error though.

**Review by hunner — APPROVED** [review]
Looks good, just needs the "wip" commit squashed or re-messaged.

File: lib/puppet/provider/iis_feature/default.rb [hunk 1]
```diff
@@ -54,29 +53,29 @@ def destroy
   def self.instances
     cmd = []
     cmd << 'Import-Module ServerManager; ' if self.is_windows2008 == true
-    cmd << 'Get-WindowsFeature | ConvertTo-Json -Depth 4'
+    cmd << 'Get-WindowsFeature | Sort Name | Select Name,Installed | ConvertTo-Json -Depth 4'
```
Comment (glennsarti) [inline lib/puppet/provider/iis_feature/default.rb:56]: Do we really need the sort, or it's just to be pretty?

File: lib/puppet_x/puppetlabs/iis/iis_features.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+module PuppetX
+  module IIS
+    module Features
```
Comment (jpogran) [inline lib/puppet_x/puppetlabs/iis/iis_features.rb:3]: Isn't it convention to have the module name the same as the file name?

File: lib/puppet/provider/iis_feature/default.rb (hunk 1 above)
Comment (jpogran) [reply lib/puppet/provider/iis_feature/default.rb:56]: Was easier to debug/pry

File: lib/puppet_x/puppetlabs/iis/iis_features.rb (hunk 2 above)
Comment (glennsarti) [reply lib/puppet_x/puppetlabs/iis/iis_features.rb:3]: Probably. but you can change all that.

**Review by glennsarti — APPROVED** [review]
LGTM

**Comment by glennsarti** [comment]
Ahdoc has passed. Merging.

**Pull request merged**